core/shared/src/main/scala/scalaz/zio/Errors.scala

```diff
@@ -6,5 +6,12 @@ object Errors {
   final case class UnhandledError(error: Any, defects: List[Throwable] = Nil)
       extends Exception("An error was not handled by a fiber: " + error.toString())
 
-  final object TerminatedFiber extends Exception("The fiber was terminated either by a defect or an interruption")
+  final case class TerminatedFiber(defect: Throwable, defects: List[Throwable])
+      extends Exception("The fiber was terminated by a defect")
+
+  final case class InterruptedFiber(causes: List[Throwable], defects: List[Throwable])
+      extends Exception("The fiber was terminated by an interruption")
+
+  final case class ParallelFiberError(left: Throwable, right: Throwable, defects: List[Throwable])
```

Comment:
As stated elsewhere, I think we can delete all these types and just have one (`FiberHalt` or something).